### PR TITLE
Do requester links

### PR DESCRIPTION
**Pull request by benoitdevos**
Finishes works of https://github.com/logion-network/logion-wallet/pull/506
Companion of https://github.com/logion-network/logion-pallets/pull/37
Companion of https://github.com/logion-network/logion-backend-ts/pull/272
___
* Refactor `LocItem` by introducing `abstract AbstractLocItem<T>`.
* Do implement (follow `todo`s) link lifecycle.
* For all items: display the correct title and reject reason when applicable.

logion-network/logion-internal#997

### CODE REVIEW

File: src/loc/LocItem.tsx [hunk 1]
```diff
@@ -123,112 +125,70 @@ export abstract class LocItem implements CommonData {
...
-        return this._fileData?.nature || "-";
-    }
-
-    override hasData(): boolean {
-        return super.hasData() && this._fileData !== undefined;
-    }
-
-    metadataData(): MetadataData {
-        throw new Error("Method not implemented.");
+        return this.data()?.nature || "-";
```
Comment (gdethier) [inline src/loc/LocItem.tsx:184]: this.data() does not return undefined or null so the ? is useless I think.

File: src/loc/LocItem.tsx [hunk 2]
```diff
@@ -240,51 +200,26 @@ export class FileItem extends LocItem {
...
-        return this._linkData?.nature || "-";
-    }
-
-    override hasData(): boolean {
-        return super.hasData() && this._linkData !== undefined;
-    }
-
-    metadataData(): MetadataData {
-        throw new Error("Method not implemented.");
+        return this.data()?.nature || "-";
```
Comment (gdethier) [inline src/loc/LocItem.tsx:206]: See above.

**Review by gdethier — APPROVED** [review]
Good job! Just a couple of comments.

File: src/loc/LocItem.tsx (hunk 1 above)
Comment (benoitdevos) [reply src/loc/LocItem.tsx]: Yse, indeed. `?` removed evrywhere it's useless.

File: src/loc/LocItem.tsx (hunk 2 above)
Comment (benoitdevos) [reply src/loc/LocItem.tsx]: Yse, indeed. `?` removed evrywhere it's useless.

**Pull request merged**